ext/pcntl/pcntl.c

```diff
@@ -1622,6 +1631,50 @@ PHP_FUNCTION(pcntl_getcpu)
 }
 #endif
 
+#if defined(HAVE_PTHREAD_SET_QOS_CLASS_SELF_NP)
+PHP_FUNCTION(pcntl_getqos_class)
+{
+	qos_class_t qos_class;
+	zend_object *qos_obj;
+
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	if (UNEXPECTED(pthread_get_qos_class_np(pthread_self(), &qos_class, NULL) != 0))
+	{
+		// unlikely unless an external tool set the QOS class with a wrong value
+		PCNTL_G(last_error) = errno;
+		zend_throw_error(NULL, "invalid QOS class %u", qos_class);
+		RETURN_THROWS();
+	}
+
+	if (UNEXPECTED(zend_enum_get_case_by_value(&qos_obj, QosClass_ce, (zend_long)qos_class, NULL, false) == FAILURE))
+	{
+		zend_throw_error(NULL, "invalid QOS value class entry %u", qos_class);
+		RETURN_THROWS();
+	}
+
+	ZVAL_COPY_OR_DUP(return_value, zend_enum_fetch_case_value(qos_obj));
```

Comment:
`zend_enum_fetch_case_value` returns the backed value, but you want the object instead.